addons/phone_validation/models/phone_blacklist.py

```diff
@@ -40,15 +45,19 @@ def create(self, values):
             done.add(sanitized)
             to_create.append(dict(value, number=sanitized))
 
-        """ To avoid crash during import due to unique email, return the existing records if any """
-        sql = '''SELECT number, id FROM phone_blacklist WHERE number = ANY(%s)'''
-        numbers = [v['number'] for v in to_create]
-        self._cr.execute(sql, (numbers,))
-        bl_entries = dict(self._cr.fetchall())
-        to_create = [v for v in to_create if v['number'] not in bl_entries]
+        # Get existing phone blacklist (for requested numbers to create)
+        numbers_requested = [tc['number'] for tc in to_create]
+        pb_existing = self.search([('active', 'in', [True, False]), ('number', 'in', numbers_requested)])
 
-        results = super(PhoneBlackList, self).create(to_create)
-        return self.env['phone.blacklist'].browse(bl_entries.values()) | results
+        # Out of existing pb records, activate non-active
+        pb_existing.filtered(lambda pb: not pb.active).write({'active': True})
+
+        # Create new records, while skipping existing_numbers
+        existing_numbers = pb_existing.mapped('number')
```

Comment:
Could be worth using a set?

Reply:
Number is unique so there won't be any duplicated entries.

Reply:
yup, why not

Reply:
soft @ tde

Meant more to avoid going through the whole list every time on the next line